Opserver.Core/Data/Dashboard/Providers/WmiDataProvider.Polling.cs

```diff
@@ -546,6 +546,32 @@ private static string GetCounterName(string original)
                 UpdateHistoryStorage(CombinedVolumePerformanceHistory, combinedUtil);
             }
 
+            public async Task<bool> UpdateServiceAsync(string serviceName, NodeService.Action action)
+            {
+                string query = $@"SELECT Name FROM Win32_Service WHERE Name = '{serviceName}'";
+
+                bool actionResult = false;
+
+                using (var q = Wmi.Query(Endpoint, query))
+                {
+                    foreach (var service in await q.GetDynamicResultAsync().ConfigureAwait(false))
+                    {
+                        switch (action)
+                        {
+                            case NodeService.Action.Stop:
+                                actionResult = service.StopService() == 0;
+                                break;
+                            case NodeService.Action.Start:
+                                actionResult = service.StartService() == 0;
+                                break;
+                            default:
+                                break;
```

Comment:
You'll want to throw an argument out of range exception here, so that in a new action is added we find and fix these kinds of things ASAP.